synapse/handlers/user_directory.py

```diff
@@ -466,9 +449,15 @@ async def _handle_possible_remote_profile_change(
         database if there are. This is intended for remote users only. The caller
         is responsible for checking that the given user is remote.
         """
-        if not prev_event_id or not event_id:
+
+        if not event_id:
             return
```

Comment:
If we wanted to simplify things, we could make `event_id` non-noneable, since we never seem to call `_handle_possible_remote_profile_change` with `event_id=None`.